crates/api_crud/src/post/create.rs

```diff
@@ -56,6 +55,7 @@ pub async fn create_post(
   let url = data_url.map(clean_url_params).map(Into::into); // TODO no good way to handle a "clear"
 
   is_valid_post_title(&data.name)?;
```

Comment:
This function already has a check for 200 char maximum length using [this regex](https://github.com/LemmyNet/lemmy/blob/main/crates/utils/src/utils/validation.rs#L8). But the regex is wrong, it should be something like `^\S{3,200}$` instead. You can change that and add keep your test cases to ensure that its working. I dont think the separate error type for invalid title length is necessary.

Reply:
Thank you. I didn't check into the regexp. Yeah, it should've been `^\S{3,200}$`.

Reply:
@Nutomic How about `^\s?.{3,200}\s?$` instead?

Reply:
Right Im not so familiar with regex shorthands. Your suggestion makes more sense, but instead of `\s?` you can just call `trim()` so that leading and trailing spaces are removed, not written to db. Also there is no reason to close this PR, you can add more commits to make these changes.